### PR TITLE
Refactor `@stdlib/blas/base/dcopy`

### PR DESCRIPTION
**Pull request by Pranavchiku**
## Checklist

- [x] update readme.md
- [x] docs
- [x] include/stdlib/math/base/special/ ( No such change visible )
- [x] src
- [x] include.gypi
- [x] manifest.json
- [x] package.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *
I will keep on populating the checklist as and when necessary.

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
I am facing some error while pushing `docs` will have a look asap.

**Comment by Pranavchiku** [comment]
I am receiving the following error while generating native addons, using `make install-node-addons NODE_ADDONS_PATTERN=blas/base/dcopy`
```console
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make[1]: Entering directory '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/blas/base/dcopy/build'
  RULE binding_gyp_addon_target_compile_fortran_linux_0 Release/obj.target/addon/geni/dcopy.o
  CC(target) Release/obj.target/addon/src/addon.o
../src/addon.c:29:1: error: unknown type name ‘namespace’
   29 | namespace stdlib_blas_base_dcopy {
      | ^~~~~~~~~
../src/addon.c:29:34: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘{’ token
   29 | namespace stdlib_blas_base_dcopy {
      |                                  ^
make[1]: *** [addon.target.mk:171: Release/obj.target/addon/src/addon.o] Error 1
make[1]: Leaving directory '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/blas/base/dcopy/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/pranavchiku/stdlib/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)
gyp ERR! System Linux 5.15.0-56-generic
gyp ERR! command "/usr/bin/node" "/home/pranavchiku/stdlib/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /home/pranavchiku/stdlib/lib/node_modules/@stdlib/blas/base/dcopy
gyp ERR! node -v v12.22.9
gyp ERR! node-gyp -v v6.1.0
gyp ERR! not ok 
Error: failed to build add-on: /home/pranavchiku/stdlib/lib/node_modules/@stdlib/blas/base/dcopy
```

**Comment by Pranavchiku** [comment]
I am not sure why it is occurring, @kgryte if you can have a look?

**Comment by kgryte** [comment]
You haven't refactored `addon.c`. C does not have namespaces. Namespaces are available in C++.

**Comment by Pranavchiku** [comment]
Ohh yes, sorry for that. On it.

**Comment by Pranavchiku** [comment]
When I try to commit changes of `index.d.ts` and `repl.txt` I receive the following log,
```console
make[1]: Entering directory '/home/pranavchiku/stdlib'

Linting file: lib/node_modules/@stdlib/blas/base/dcopy/docs/types/index.d.ts
Error: Cannot find module '/home/pranavchiku/.dts/typescript-installs/4.2/node_modules/typescript'
Require stack:
- /home/pranavchiku/stdlib/node_modules/dtslint/bin/lint.js
- /home/pranavchiku/stdlib/node_modules/dtslint/bin/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at testDependencies (/home/pranavchiku/stdlib/node_modules/dtslint/bin/lint.js:64:16)
    at Object.<anonymous> (/home/pranavchiku/stdlib/node_modules/dtslint/bin/lint.js:26:28)
    at Generator.next (<anonymous>)
    at /home/pranavchiku/stdlib/node_modules/dtslint/bin/lint.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/home/pranavchiku/stdlib/node_modules/dtslint/bin/lint.js:4:12)
make[1]: *** [/home/pranavchiku/stdlib/tools/make/lib/lint/typescript/dtslint.mk:110: dtslint-files] Error 1
make[1]: Leaving directory '/home/pranavchiku/stdlib'
make: *** [/home/pranavchiku/stdlib/tools/make/lib/lint/typescript/Makefile:178: lint-typescript-declarations-files] Error 2

TypeScript declaration file lint errors.
```

**Comment by kgryte** [comment]
Try deleting your node modules folder and doing a fresh `make install-node-modules && make init`.

**Comment by Pranavchiku** [comment]
Nope, I still receive same error.

**Comment by kgryte** [comment]
Are you able to confirm that the `.dts` folder exists, as indicated in the error message?

**Comment by Pranavchiku** [comment]
It does not exist.
Shouldn't it be there after `make init` ?

**Comment by kgryte** [comment]
No. `dtslint` should automatically create it. What version of `dtslint` do you have installed?

**Comment by Pranavchiku** [comment]
I guess I don't have `dtslint` installed.
If I do
```console
$ dtslint --version
Command 'dtslint' not found, did you mean:
  command 'dtblint' from deb dt-utils (2021.03.0+ds-2)
Try: sudo apt install <deb name>
```

**Comment by kgryte** [comment]
No, it would be in your `node_modules` folder.

**Comment by Pranavchiku** [comment]
I am not sure about version, `dtslint` is present in `node_modules` and I recently did `npm install --save-dev dtslint` by looking at readme.

Edit: `dtslint@3.7.0`

**Comment by kgryte** [comment]
Okay, that is the same version as me, and mine currently works, so not sure why your installation is throwing the error above.

**Comment by kgryte** [comment]
What version of Node.js are you running?

**Comment by Pranavchiku** [comment]
>What version of Node.js are you running?

v12.22.9

**Comment by kgryte** [comment]
Try upgrading to at least `v16` (e.g., `nvm install 16`).

**Comment by kgryte** [comment]
You can also try to lint the files directly via

```bash
$ make lint-typescript-declarations TYPESCRIPT_DECLARATIONS_FILTER=.*/blas/base/dcopy/.*
```

**Comment by Pranavchiku** [comment]
> Try upgrading to at least v16 (e.g., nvm install 16).

I did this, now trying `make install-node-modules && make init` let's hope it works this time.

**Comment by Pranavchiku** [comment]
Still facing same issue. I guess `.dts`. folder is not present in my device, not sure why.

**Comment by kgryte** [comment]
A bit odd that you wouldn't have encountered this error before.

**Comment by Pranavchiku** [comment]
Yeah, I didn't encounter updating `docs` in past PRs although I did those in my early PRs which had different Ubuntu settings as I did a clean installation in month of december.

**Comment by Pranavchiku** [comment]
Apart from `index.d.ts` 
I tried to make the necessary changes, and all tests and benchmarks are cleared.

**Comment by kgryte** [comment]
Maybe you have a permissions issue. You can try manually creating a `.dts` folder in your home directory with suitable permissions.

**Comment by kgryte** [comment]
You can also try doing some `console.log`ging in your local node_modules folder in the package responsible for installing TypeScript versions: <https://github.com/microsoft/DefinitelyTyped-tools/blob/master/packages/utils/src/typescript-installer.ts>

It's possible that errors are being swallowed regarding permissions, etc.

**Comment by Pranavchiku** [comment]
I have the folder installed, it does not have `node_modules` in the path, `/home/pranavchiku/.dts/typescript-installs/4.2/node_modules/typescript`.

**Comment by kgryte** [comment]
In short, I don't have an easy answer atm, apart from doing a bit of digging and debugging on your end.

**Comment by Pranavchiku** [comment]
No worries, I will try to get it done asap.

**Comment by kgryte** [comment]
My initial sense is to start putting console log calls in the code actually responsible for the installations. Mucking about with the `.dts` folder is likely not where the problem is.

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared. This PR can have a review.

Apologies for taking so long to complete this PR.

File: lib/node_modules/@stdlib/blas/base/dcopy/docs/repl.txt [hunk 1]
```diff
@@ -22,15 +22,15 @@
...
+        Output array.
 
     strideY: integer
         Index increment for `y`.
 
     Returns
     -------
     y: Float64Array
-        Input array `y`.
+        Outpue array.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dcopy/docs/repl.txt]:
```suggestion
        Output array.
```

File: lib/node_modules/@stdlib/blas/base/dcopy/src/addon.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/base/dcopy.h"
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_double.h"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dcopy/src/addon.c]:
```suggestion
```

This isn't used.

File: lib/node_modules/@stdlib/blas/base/dcopy/manifest.json [hunk 3]
```diff
@@ -112,7 +142,13 @@
 			],
 			"libraries": [],
 			"libpath": [],
-			"dependencies": []
+			"dependencies": [
+				"@stdlib/napi/export",
+				"@stdlib/napi/argv",
+				"@stdlib/napi/argv-double",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dcopy/manifest.json]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/base/dcopy/manifest.json [hunk 4]
```diff
@@ -99,7 +123,13 @@
 				"-lpthread"
 			],
 			"libpath": [],
-			"dependencies": []
+			"dependencies": [
+				"@stdlib/napi/export",
+				"@stdlib/napi/argv",
+				"@stdlib/napi/argv-double",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dcopy/manifest.json]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/base/dcopy/manifest.json [hunk 5]
```diff
@@ -83,7 +101,13 @@
 				"-lblas"
 			],
 			"libpath": [],
-			"dependencies": []
+			"dependencies": [
+				"@stdlib/napi/export",
+				"@stdlib/napi/argv",
+				"@stdlib/napi/argv-double",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dcopy/manifest.json]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/base/dcopy/manifest.json [hunk 6]
```diff
@@ -68,7 +80,13 @@
 			],
 			"libraries": [],
 			"libpath": [],
-			"dependencies": []
+			"dependencies": [
+				"@stdlib/napi/export",
+				"@stdlib/napi/argv",
+				"@stdlib/napi/argv-double",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dcopy/manifest.json]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/base/dcopy/manifest.json [hunk 7]
```diff
@@ -54,7 +60,13 @@
 				"-lpthread"
 			],
 			"libpath": [],
-			"dependencies": []
+			"dependencies": [
+				"@stdlib/napi/export",
+				"@stdlib/napi/argv",
+				"@stdlib/napi/argv-double",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dcopy/manifest.json]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/base/dcopy/manifest.json [hunk 8]
```diff
@@ -38,7 +38,13 @@
 			],
 			"libraries": [],
 			"libpath": [],
-			"dependencies": []
+			"dependencies": [
+				"@stdlib/napi/export",
+				"@stdlib/napi/argv",
+				"@stdlib/napi/argv-double",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/dcopy/manifest.json]:
```suggestion
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Pull request merged**